components/raftstore/src/store/worker/read.rs

```diff
@@ -782,7 +782,6 @@ where
                     // Replica can serve stale read if and only if its `safe_ts` >= `read_ts`
                     RequestPolicy::StaleRead => {
                         let read_ts = decode_u64(&mut req.get_header().get_flag_data()).unwrap();
-                        assert!(read_ts > 0);
```

Comment:
We need to make sure it's OK to remove the assert.

Reply:
Without setting is_snap_for_gc in SnapContext, stale read behaves the same with before.

Reply:
I would suggest to somehow pass is_snap_for_gc to here and keep the assert when is_snap_for_gc is false/

Reply:
> I would suggest to somehow pass is_snap_for_gc to here and keep the assert when is_snap_for_gc is false/

Good suggestion.